### PR TITLE
fix: fix ordering of logic for checking persisting and loading erc721 nfts

### LINKED ISSUE #1564: ️🛠 - Add migration function for `persistedNfts`
### Task description

This store needs a migration function to adjust existing data according to new interfaces.

### Requirements

_TBD_

### LINKED ISSUE #1582: ️🛠 - Fix ordering of logic for checking, persisting, and loading ERC721 NFTs
### Task description

Currently the logic in login function is in such an order that the loading of NFTs happens before the logic to check and persist them, so the first time a profile is logged into its ERC721 NFTs will **NOT** be loaded.

### Requirements

_TBD_

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/src/lib/core/nfts/actions/addOrUpdateNftInAllAccountNfts.ts [hunk 1]
```diff
@@ -17,3 +17,19 @@ export function addOrUpdateNftInAllAccountNfts(accountIndex: number, ...newNfts:
...
+    allAccountNfts.update((state) => {
+        for (const accountNfts of Object.values(state)) {
+            for (const newNft of newNfts) {
+                const nft = accountNfts.find((_nft) => _nft.id === newNft.id)
+                if (nft) {
+                    Object.assign(nft, newNft)
+                } else {
+                    accountNfts.push(newNft)
+                }
+            }
```
Comment (Tuditi) [inline packages/shared/src/lib/core/nfts/actions/addOrUpdateNftInAllAccountNfts.ts]: This logic has been duplicated in the function above. This function has a confusing name as well. Maybe `addOrUpdateAllAccountNfts` would do the job?

**Review by Tuditi — CHANGES_REQUESTED** [review]
Left a small request for change. Otherwise works excellent! :muscle:

File: packages/shared/src/lib/core/nfts/actions/addOrUpdateNftInAllAccountNfts.ts [hunk 2]
```diff
@@ -17,3 +17,19 @@ export function addOrUpdateNftInAllAccountNfts(accountIndex: number, ...newNfts:
         return state
     })
 }
+
+export function addOrUpdateAllAccountNfts(...newNfts: INft[]): void {
```
Comment (nicole-obrien) [inline packages/shared/src/lib/core/nfts/actions/addOrUpdateNftInAllAccountNfts.ts]: updateAllAccountNfts for this one

updateAllAccountNftsForAccount for the one above

**Review by Tuditi — DISMISSED** [review]
Works for me, nice job! :muscle:

**Review by Tuditi — APPROVED** [review]
LGTM!

**Pull request merged**